### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by rociac**
Hello, looking forward to your feedback!

### CODE REVIEW

**Comment by rociac** [comment]
We fixed some spacing errors on the linter but we don't know if we should change the others, looking forward for your feedback!

**Comment by AdnanAfsari** [comment]
### **ADNAN AFSARI WORKING ON IT**

File: ruby.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+def bubble_sort(array)
+  need_sort = true
+  while need_sort
+    need_sort = false
+    for i in 0...array.length - 1
```
Comment (AdnanAfsari) [inline ruby.rb]: let me ask seniors about this kind of errors in this project.

File: ruby.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      end
+    end
+  end
+end
+
+def bubble_sort_by(array)
+  need_sort = true
+  while need_sort
+    need_sort = false
+    for i in 0...array.length - 1
```
Comment (AdnanAfsari) [inline ruby.rb]: let me ask seniors about this kind of errors in this project.

**Review by AdnanAfsari — CHANGES_REQUESTED** [review]
Hello @allun26 @Frodoosk 

Project looks good so far but you have to fix those linter errors. there are two errors that i commented i am also not sure in this project. i will let you know in your future review.
but fix other errors and submit for another review.

you can check this repo if you need any help with linter 
https://github.com/microverseinc/linters-config#linters-config

Happy Coding

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Nice job @allun26 , @Frodoosk 
The logic is right and the test case have passed. Still the same errors marked by my fellow TSE appear in the checks.

**`Prefer each over for.`**
We can solve it two ways:
- `array[0..-2].each.with_index do |_, i|`
  
  - Note that we excluded that last element with `[0..-2]` because `array[i + 1]` will give error at the last loop iteration.
  - The `_` in the `|_, i|` because the element will not be used inside the array so that way we ignore it. Here's a link from [ruby style guidelines](https://github.com/rubocop-hq/ruby-style-guide#trailing-underscore-variables).

- `(array.length - 1).times do |i|` This also won't give errors

Please fix these minor issues and submit a new code review.

### Happy Coding :computer:

**Comment by mosaaleb** [comment]
### Congratulations @allun26 @Frodoosk :clap:

You can merge into master :+1: 

Good luck with the next projects.

## Happy Coding :computer:

**Review by mosaaleb — APPROVED** [review]
### Congratulations @allun26 @Frodoosk :clap:

You can merge into master :+1: 

Good luck with the next projects.

## Happy Coding :computer:

**Pull request merged**